common/djangoapps/track/middleware.py

```diff
@@ -196,6 +198,15 @@ def get_username(self, request):
         except AttributeError:
             return ''
 
+    def get_request_ip(self, request):
+        """Gets the IP address of the request"""
+        ip = get_ip(request)
+        if ip is not None:
+            return ip
+        else:
+            return ''
```

Comment:
`None` in this case is not a failure, just that there was no IP header to be pulled.  To be consistent with how all of the other event attributes are populated we fill it with an empty string rather than `None`.